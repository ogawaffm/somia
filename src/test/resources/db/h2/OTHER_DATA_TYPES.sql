DROP TABLE IF EXISTS TEST.OTHER_DATA_TYPES;
CREATE TABLE TEST.OTHER_DATA_TYPES
(

    BOOLEAN_COL                BOOLEAN,
    UUID_COL                   UUID,

    JAVA_OBJECT_COL            JAVA_OBJECT,
    JAVA_OBJECT_1_COL          JAVA_OBJECT(1),
    JAVA_OBJECT_1000000000_COL JAVA_OBJECT(1000000000),

    OBJECT_COL                 OBJECT,
    OBJECT_1_COL               OBJECT(1),
    OBJECT_1000000000_COL      OBJECT(1000000000),

    OTHER_COL                  OTHER,
    OTHER_1_COL                OTHER(1),
    OTHER_1000000000_COL       OTHER(1000000000),

    JSON_COL                   JSON,
    JSON_1_COL                 JSON(1),
    JSON_1000000000_COL        JSON(1000000000),

    ENUM_A_B_C_COL             ENUM ('A', 'B', 'C')

    -- H2 does not support XML data type

)