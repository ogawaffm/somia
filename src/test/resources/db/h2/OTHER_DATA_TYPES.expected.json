{
  "name": "OTHER_DATA_TYPES",
  "catalogName": "TEST",
  "schemaName": "PUBLIC",
  "typeName": "BASE TABLE",
  "comment": "",
  "columns": [
    {
      "name": "BOOLEAN_COL",
      "type": {
        "name": "BOOLEAN",
        "typeNumber": 16,
        "className": "java.lang.Boolean",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 5,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "UUID_COL",
      "type": {
        "name": "UUID",
        "typeNumber": -2,
        "className": "byte[]",
        "precision": 16,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 32,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "JAVA_OBJECT_COL",
      "type": {
        "name": "JAVA_OBJECT",
        "typeNumber": 2000,
        "className": "java.lang.Object",
        "precision": 1000000000,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": false,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 2147483647,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "JAVA_OBJECT_1_COL",
      "type": {
        "name": "JAVA_OBJECT",
        "typeNumber": 2000,
        "className": "java.lang.Object",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": false,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 2147483647,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "JAVA_OBJECT_1000000000_COL",
      "type": {
        "name": "JAVA_OBJECT",
        "typeNumber": 2000,
        "className": "java.lang.Object",
        "precision": 1000000000,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": false,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 2147483647,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "OBJECT_COL",
      "type": {
        "name": "JAVA_OBJECT",
        "typeNumber": 2000,
        "className": "java.lang.Object",
        "precision": 1000000000,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": false,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 2147483647,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "OBJECT_1_COL",
      "type": {
        "name": "JAVA_OBJECT",
        "typeNumber": 2000,
        "className": "java.lang.Object",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": false,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 2147483647,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "OBJECT_1000000000_COL",
      "type": {
        "name": "JAVA_OBJECT",
        "typeNumber": 2000,
        "className": "java.lang.Object",
        "precision": 1000000000,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": false,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 2147483647,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "OTHER_COL",
      "type": {
        "name": "JAVA_OBJECT",
        "typeNumber": 2000,
        "className": "java.lang.Object",
        "precision": 1000000000,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": false,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 2147483647,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "OTHER_1_COL",
      "type": {
        "name": "JAVA_OBJECT",
        "typeNumber": 2000,
        "className": "java.lang.Object",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": false,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 2147483647,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "OTHER_1000000000_COL",
      "type": {
        "name": "JAVA_OBJECT",
        "typeNumber": 2000,
        "className": "java.lang.Object",
        "precision": 1000000000,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": false,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 2147483647,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "JSON_COL",
      "type": {
        "name": "JSON",
        "typeNumber": 1111,
        "className": "java.lang.Object",
        "precision": 1000000000,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "JSON_1_COL",
      "type": {
        "name": "JSON",
        "typeNumber": 1111,
        "className": "java.lang.Object",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "JSON_1000000000_COL",
      "type": {
        "name": "JSON",
        "typeNumber": 1111,
        "className": "java.lang.Object",
        "precision": 1000000000,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "ENUM_A_B_C_COL",
      "type": {
        "name": "ENUM('A', 'B', 'C')",
        "typeNumber": 1111,
        "className": "java.lang.Object",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    }
  ],
  "indexes": [],
  "collation": "",
  "primaryKey": null,
  "bestRowIdentifier": null
}